File: backend/order-service/src/main/java/com/grocery/orderservice/service/OrderService.java
package com.grocery.orderservice.service;

import com.grocery.orderservice.dto.OrderRequestDto;
import com.grocery.orderservice.dto.OrderResponseDto;

import java.util.List;

public interface OrderService {
    OrderResponseDto createOrder(OrderRequestDto requestDto);
    OrderResponseDto getOrderById(Long orderId);
    List<OrderResponseDto> getAllOrders();
    OrderResponseDto updateOrder(Long orderId,OrderRequestDto requestDto);
    OrderResponseDto updateOrderStatus(Long orderId,String status);
    void deleteOrder(Long orderId);
    void cancelOrder(Long orderId);

    OrderResponseDto confirmOrder(Long orderId);
}

File: frontend/src/services/orderService.js
import api from "./api";

export const getAllOrders = () => {
  return api.get("http://localhost:8084/api/orders");
};

export const confirmOrder = (orderId) => {
  return api.put(`http://localhost:8084/api/orders/${orderId}/confirm`);
};

export const updateOrderStatus = (orderId, status) => {
  return api.patch(
    `http://localhost:8084/api/orders/${orderId}/status?status=${status}`,
  );
};

export const cancelOrder = (orderId) => {
  return api.put(`http://localhost:8084/api/orders/${orderId}/cancel`);
};
